cmake/sca/eclair/sca.cmake

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, BUGSENG Srl
+
+find_program(ECLAIR_ENV eclair_env REQUIRED)
+message(STATUS "Found eclair_env: ${ECLAIR_ENV}")
+
+find_program(ECLAIR_REPORT eclair_report REQUIRED)
+message(STATUS "Found eclair_report: ${ECLAIR_REPORT}")
+
+# ECLAIR Settings
+set(ECLAIR_PROJECT_NAME "Zephyr-${BOARD}")
+set(ECLAIR_OUTPUT_DIR ${CMAKE_BINARY_DIR}/sca/eclair_out)
+set(ECLAIR_ECL_DIR "${ZEPHYR_BASE}/cmake/sca/eclair/ECL")
+set(ECLAIR_DIAGNOSTICS_OUTPUT "${ECLAIR_OUTPUT_DIR}/DIAGNOSTIC.txt")
+set(ECLAIR_DATA_DIR "${ECLAIR_OUTPUT_DIR}/data")
+set(ECLAIR_BUILD_DIR "${ECLAIR_OUTPUT_DIR}/build")
+set(ECLAIR_PROJECT_ECD "${ECLAIR_OUTPUT_DIR}/PROJECT.ecd")
+set(CC_ALIASES "${CMAKE_C_COMPILER}")
+set(CXX_ALIASES "${CMAKE_CXX_COMPILER}")
+set(AS_ALIASES "${CMAKE_AS}")
+set(LD_ALIASES "${CMAKE_LINKER}")
+set(AR_ALIASES "${CMAKE_ASM_COMPILER_AR} ${CMAKE_C_COMPILER_AR} ${CMAKE_CXX_COMPILER_AR}")
+if(NOT ECLAIR_RULES_SET)
+  set(ECLAIR_RULES_SET first_analysis)
+endif()
+
+# Create Output Directory
+file(REMOVE_RECURSE ${ECLAIR_OUTPUT_DIR})
+file(MAKE_DIRECTORY ${ECLAIR_OUTPUT_DIR})
+file(MAKE_DIRECTORY ${ECLAIR_DATA_DIR})
+message(STATUS "ECLAIR outputs have been written to: ${ECLAIR_OUTPUT_DIR}")
+message(STATUS "ECLAIR ECB files have been written to: ${ECLAIR_DATA_DIR}")
+message(STATUS "ECLAIR BUILD DIR is: ${ECLAIR_BUILD_DIR}")
+
+add_custom_target(eclair ALL
+  COMMAND sh -c "ECLAIR_PROJECT_NAME=${ECLAIR_PROJECT_NAME} \
+    ECLAIR_PROJECT_ROOT=${ZEPHYR_BASE} \
+    ECLAIR_DIAGNOSTICS_OUTPUT=${ECLAIR_DIAGNOSTICS_OUTPUT} \
+    ECLAIR_DATA_DIR=${ECLAIR_DATA_DIR} \
+    ECLAIR_RULES_SET=${ECLAIR_RULES_SET} \
+    CC_ALIASES=\"${CC_ALIASES}\" \
+    CXX_ALIASES=\"${CXX_ALIASES}\" \
+    AS_ALIASES=\"${AS_ALIASES}\" \
+    LD_ALIASES=\"${LD_ALIASES}\" \
+    AR_ALIASES=\"${AR_ALIASES}\" \
+    ${ECLAIR_ENV} \
+    -verbose \
+    -eval_file=${ECLAIR_ECL_DIR}/analysis.ecl \
+    -- west build -p always \
+                  -b ${BOARD} \
+                  --build-dir ${ECLAIR_BUILD_DIR} \
+                  ${APPLICATION_SOURCE_DIR} \
+                  | tee ${ECLAIR_OUTPUT_DIR}/analysis.log"
```

Comment:
Are you referring to the tee?

The tee is not related to ECLAIR (or to the analysis results), it just saves the west invocation logs in
analysis.log. I'm fine with removing it. In case of problems, we still will have something reported in
$ECLAIR_DIAGNOSTICS_OUTPUT.

Doing that, if the build fails it will result in an ECLAIR error in $ECLAIR_DIAGNOSTICS_OUTPUT,
but the actual build error will not be present (maybe this is not even necessary, maybe west already
put these logs somewhere).

Since a broken build is not related to ECLAIR I'm completely fine in removing the tee.